src/backend/binaryen/wasm_expr_gen.ts

```diff
@@ -2765,43 +2750,70 @@ export class WASMExpressionGen {
             infcPropTypeIdRef,
             propTypeIdRef,
         );
-        /* Hint for benchmark: here we add a comparation between two prop types, and an additional call for wasm function is called */
-        /* TODO: reduce the additional call */
-        /* if two prop type id is equal, we can use the prop type information get from compiler time */
-        const ifPropTypeIdEqualTrue = this.getIndirectValueRef(
+        let ifPropTypeIdEqualTrue = this.getIndirectValueRef(
             objRef,
             indexRef,
             flagRef,
             valueType,
         );
-        const anyTypedProperty = this.module.call(
-            UtilFuncs.getFuncName(
-                BuiltinNames.builtinModuleName,
-                BuiltinNames.getPropertyIfTypeIdMismatch,
-            ),
-            [
+        if (isOptional) {
+            ifPropTypeIdEqualTrue = this.module.if(
+                FunctionalFuncs.isPropertyUnExist(this.module, flagAndIndexRef),
+                FunctionalFuncs.generateDynUndefined(this.module),
+                FunctionalFuncs.boxNonLiteralToAny(
+                    this.module,
+                    ifPropTypeIdEqualTrue,
+                    valueType.kind,
+                ),
+            );
+        }
+        /* Hint for benchmark: here we add a comparation between two prop types, and an additional call for wasm function is called */
+        /* TODO: reduce the additional call */
+        /* if two prop type id is equal, we can use the prop type information get from compiler time */
+        let ifPropTypeIdEqualFalse: binaryen.ExpressionRef;
+        if (
+            valueType instanceof UnionType ||
+            (valueType instanceof FunctionType && isOptional)
+        ) {
+            /* For union type, we can parse type one by one to avoid runtime getting */
+            /* Hint: only optional function will be regarded as the function type, otherwise it will be union type */
```

Comment:
Had better use more precise description, e.g.:
``` markdown
For type T? (optional<T>):
    - when T is function type, it's regarded as T
    - otherwise it's regarded as union (T | undefined)
```